File: apps/api/src/app/workout/workout.controller.ts
import { Body, Controller, Delete, Get, Param, Post, Query, UseGuards } from '@nestjs/common';
import { AuthGuard } from '../auth/auth.guard';
import { WorkoutService } from './workout.service';
import { Workout } from '../../../../../libs/interfaces/workout';
import { CurrentUser } from '../auth/current-user-decorator';
import { User } from '../../../../../libs/interfaces/user';

@UseGuards(AuthGuard)
@Controller('workout')
export class WorkoutController {

    constructor(
        private readonly workoutService: WorkoutService
    ) {
    }

    @Post()
    create(@CurrentUser() user: User, @Body() workout: Workout) {
        workout.user = user;
        return this.workoutService.create(workout);
    }

    @Get('find-by-date')
    findByDate(@CurrentUser() user: User, @Query('start') start: string, @Query('end') end: string) {
        return this.workoutService.findByDate(user.id, new Date(start), new Date(end));
    }

    @Get()
    find(@CurrentUser() user: User) {
        return this.workoutService.find(user.id);
    }

    @Get('count-total-days-workout')
    countTotalDaysWorkout(@CurrentUser() user: User) {
        return this.workoutService.countTotalDaysWorkout(user.id);
    }

    @Get(':id')
    findById(@Param('id') id: string) {
        return this.workoutService.findById(Number(id));
    }

    @Delete()
    delete(@CurrentUser() user: User, @Query('id') id: number) {
        return this.workoutService.delete(user.id, Number(id));
    }
}

File: my-workout-api/src/main/java/com/avernet/myworkoutapi/workout/WorkoutController.java
package com.avernet.myworkoutapi.workout;

import com.avernet.myworkoutapi.user.UserEntity;
import jakarta.annotation.Resource;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("workout")
public class WorkoutController {

    @Resource
    WorkoutService workoutService;

    @PostMapping
    Workout create(@AuthenticationPrincipal UserEntity userEntity, @RequestBody WorkoutRequest workoutRequest) {
        return workoutService.createWorkout(userEntity, workoutRequest);
    }

    @GetMapping("{id}")
    WorkoutGroupedHistories find(@PathVariable Long id) {
        return workoutService.find(id);
    }

    @GetMapping("find-by-date")
    List<Workout> findByDate(@AuthenticationPrincipal UserEntity userEntity, @RequestParam String start, @RequestParam String end) {
        return workoutService.findByDate(userEntity, start, end);
    }

    @GetMapping("count-total-days-workout")
    Integer countTotalDaysWorkout(@AuthenticationPrincipal UserEntity userEntity) {
        return workoutService.countTotalDaysWorkout(userEntity);
    }

    @DeleteMapping("{id}")
    void delete(@AuthenticationPrincipal UserEntity userEntity, @PathVariable Long id) {
        workoutService.delete(userEntity, id);
    }
}
